modules/infinite-scroll/infinity.php

```diff
@@ -249,9 +249,18 @@ static function get_settings() {
 	 * @return int
 	 */
 	static function posts_per_page() {
-		$posts_per_page = self::get_settings()->posts_per_page ? self::get_settings()->posts_per_page : self::wp_query()->get( 'posts_per_page' );
+		$posts_per_page             = self::get_settings()->posts_per_page ? self::get_settings()->posts_per_page : self::wp_query()->get( 'posts_per_page' );
+		$posts_per_page_core_option = get_option( 'posts_per_page' );
+
+		// If Infinite Scroll is set to click, and if the site owner changed posts_per_page, let's use that.
+		if (
+			'click' === self::get_settings()->type
+				&& ( '10' !== $posts_per_page_core_option )
+		) {
+			$posts_per_page = $posts_per_page_core_option;
+		}
 
-		// Take JS query into consideration here
+		// Take JS query into consideration here.
```

Comment:
Changes in this file (contained in a single commit) look like they shouldn't be in this PR, right?

Reply:
Sorry, yes that's a mistake. I've removed those commits.